### PR TITLE
feat: separate stubs into their own wheel

### PR DESCRIPTION
**Pull request by daniel-makerx**

### CODE REVIEW

File: stubs/README.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# AlgoPy - API for writing Algorand Python smart contracts
+
+## Installation
+
+### API
+The minimum supported Python version for writing Algorand Python smart contracts is 3.12.
+
+You can install AlgoPy from PyPI into your project virtualenv:
+```shell
+pip install algopy-stubs
```
Comment (robdmoore) [inline stubs/README.md:10]:
```suggestion
pip install algorand-python
```

File: stubs/README.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+```
+If you're using poetry for dependency and virutalenv management, you can add it that way with
+`poetry add algopy-stubs`.
```
Comment (robdmoore) [inline stubs/README.md:13]:
```suggestion
poetry add algorand-python
```

File: stubs/README.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
+# AlgoPy - API for writing Algorand Python smart contracts
```
Comment (robdmoore) [inline stubs/README.md:1]:
```suggestion
# Algorand Python framework - API for writing Algorand Python smart contracts
```

File: stubs/README.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+## Installation
+
+### API
+The minimum supported Python version for writing Algorand Python smart contracts is 3.12.
+
+You can install AlgoPy from PyPI into your project virtualenv:
```
Comment (robdmoore) [inline stubs/README.md:8]:
```suggestion
You can install the Algorand Python framework from PyPI into your project virtualenv:
```

File: stubs/README.md (hunk 3 above)
Comment (robdmoore) [reply stubs/README.md:1]: AlgoPy isn't the name of it, we don't want to have confusion by introducing extra terminology, `algopy` is simply the abbreviated import path for Algorand Python.

File: stubs/README.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+as a dev dependency in your project virtualenv (e.g. `poetry add puya --group=dev`)
+
+## Usage
+
+To compile a contract or contracts, just supply the path(s) - either to the .py files themselves,
+or the containing directories. In the case of containing directories, any contracts discovered
+therein will be compiled, allowing you to compile multiple contracts at once. You can also supply
+more than one path at a time to the compiler.
+
+e.g. `puyapy my_project/` or `puyapy my_project/contract.py` will work to compile a single contract.
```
Comment (robdmoore) [inline stubs/README.md:27]: This shouldn't be here, it's covered in the other documentation. Maybe we can have a Usage one liner that links back to the language guide and/or main readme?

File: stubs/pyproject.toml [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+    { include = "algopy-stubs", from = "src" },
+]
+classifiers = [
+    "Development Status :: 5 - Production/Stable",
+    "Intended Audience :: Developers",
+    "Typing :: Stubs Only",
+]
+
+[tool.poetry.dependencies]
+python = "^3.11"
```
Comment (robdmoore) [inline stubs/pyproject.toml:20]: Should we just make this 3.12?

File: stubs/README.md (hunk 3 above)
Comment (robdmoore) [reply stubs/README.md:1]: unsure if we call it "framework" or "APIs"?

File: stubs/src/algopy.py [hunk 7]
```diff
@@ -0,0 +1,10 @@
+import typing
+
+if not typing.TYPE_CHECKING:
+    raise RuntimeError(
+        "`import algopy` cannot be executed by the python interpreter. "
+        "It is intended to be used by the puyapy compiler to compile "
+        "Algorand Smart Contracts written in Python to TEAL. "
```
Comment (robdmoore) [inline stubs/src/algopy.py:7]:
```suggestion
        "Algorand smart contracts written in Algorand Python. "
```

File: stubs/src/algopy.py [hunk 8]
```diff
@@ -0,0 +1,10 @@
...
+        "For more details see "
+        "https://algorandfoundation.github.io/puya/index.html#compiler-usage"
```
Comment (robdmoore) [inline stubs/src/algopy.py:9]: hmm, we should probably get this repository renamed from puya to python...

File: stubs/README.md [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+### API
+The minimum supported Python version for writing Algorand Python smart contracts is 3.12.
+
+You can install AlgoPy from PyPI into your project virtualenv:
+```shell
+pip install algopy-stubs
+```
+If you're using poetry for dependency and virutalenv management, you can add it that way with
+`poetry add algopy-stubs`.
+
```
Comment (robdmoore) [inline stubs/README.md:14]:
```suggestion
Once you have installed the Algorand Python framework, you can access the type definitions from the `algopy` module, e.g.

\`\`\`python
from algopy import Contract
\`\`\`
```

**Review by robdmoore — APPROVED** [review]

File: stubs/README.md (hunk 3 above)
Comment (achidlow) [reply stubs/README.md:1]: It's both, but framework is the broader term so that gets my +1

**Pull request merged**